src/test/java/com/google/devtools/build/lib/remote/GrpcRemoteCacheTest.java

```diff
@@ -1016,4 +1016,67 @@ public void read(ReadRequest request, StreamObserver<ReadResponse> responseObser
       assertThat(st.getCode()).isEqualTo(Status.Code.DEADLINE_EXCEEDED);
     }
   }
+
+  @Test
+  public void isRemoteCacheOptionsWhenGrpcEnabled() {
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.remoteCache = "grpc://some-host.com";
+
+    assertThat(GrpcRemoteCache.isRemoteCacheOptions(options)).isTrue();
+  }
+
+  @Test
```

Comment:
Consider merging all these into one method. I think it's a lot of boiler plate given that the test cases are so short. I think this looks much better, no?

```java
@Test
public void testIsRemoteCacheOptions() {
  RemoteOptions options = Options.getDefaults(RemoteOptions.class);
  assertThat(GrpcRemoteCache.isRemoteCacheOptions(options)).isFalse();

  options.remoteCache = "grpcs://some-host.com";
  assertThat(GrpcRemoteCache.isRemoteCacheOptions(options)).isTrue();

  options.remoteCache = "some-host.com";
  assertThat(GrpcRemoteCache.isRemoteCacheOptions(options)).isTrue();

  options.remoteCache = "http://some-host.com";
  assertThat(GrpcRemoteCache.isRemoteCacheOptions(options)).isFalse();

  options.remoteCache = "HTTP://some-host.com";
  assertThat(GrpcRemoteCache.isRemoteCacheOptions(options)).isFalse();

  options.remoteCache = "https://some-host.com";
  assertThat(GrpcRemoteCache.isRemoteCacheOptions(options)).isFalse();

  options.remoteCache = "";
  assertThat(GrpcRemoteCache.isRemoteCacheOptions(options)).isFalse();
}

Reply:
Maybe, but I would argue that in this case 1) we would test several scenarios in one method and 2) different test cases will be dependent. And if the first line will fail - we will no know until we fix that whether the 10th line also broken or not.  I personally do not like that and generally it is recommended to avoid such tests. So I'd keep the current implementation (it is already done anyways).